bcgov/biohubbc
# BHBC-2274: Router technical debt

# Overview

Fixes some minor annoyances + addresses some technical debt surrounding the app router

## Links to Jira tickets

 - https://quartech.atlassian.net/browse/BHBC-2274

## Description of relevant changes

 - Replaces normal anchor links with React Router links
 - Removes unneccessary routes:
    - `surveys/:survey_id/attachments`
    - `surveys/:survey_id/observations`
    - `surveys/:survey_id/results`
 - Removes `<Layout>` components that are already wrapped

 - Fixes an issue with the Forbidden page, wherein users cannot navigate away from the forbidden page if the initial request to fetch the user's userdata fails. This often occurs during development when the app refreshes while the API is momentarily re-compiling.

## Testing Procedures

0. Create a project and survey.
1. Click into the project, then click into the survey. It should:

 - [ ] Fetch and render the survey immediately without reloading the webpage

2. From any page in SIMS, reload the browser _while_ the API is rebuilding. You can achive this by going to `api/src/app.ts` in your IDE  and saving the file, triggering a rebuild of the API. If this is successful, you should land on the Forbidden page in SIMS.
3. Wait until the API is finished rebuilding, then attempt to click away from the Forbidden page. You should:

 - [ ] Successfully navigate away from the forbidden page
